endaq/batch/core.py

```diff
@@ -206,14 +207,16 @@ def _make_peak_windows(ch_data_cache: analyzer.CalcCache, margin_len):
         )
     )
 
+    # minor version of python 3.x
+    minor_version = sys.version_info.minor
+
     # Format results
-    result = (
-        aligned_peak_data.stack()
-        .stack()
-        .reorder_levels(["axis", "peak time", "peak offset"])
-    )
+    # Use new implementation of future_stack if Python version >= 3.9
+    levels = ["axis", "peak time", "peak offset"]
+    if minor_version < 9:
```

Comment:
```suggestion
    if sys.version_info< (3, 9):
```
I know this won't run in Python 2.x, and Python 4 isn't even on the horizon, but this is easy and maybe more obvious what's happening.